lib/hsa/kernels/upchannelize_flip_hfb.cl

```diff
@@ -0,0 +1,309 @@
+#define PI2 1.5707963267948966f
+#define PI4 0.7853981633974483f
+#define PI8 0.39269908169872414f
+#define PI16 0.19634954084936207f
+#define PI32 0.09817477042468103f
+#define PI64 0.04908738521234052f
+#define REAL    x
+#define IMAG    y
+#define CHUNK_OFFSET 128
+#define n_all 256
+#define scaling 400.
+
+__constant float BP[16] = { 0.52225748 , 0.58330915 , 0.6868705 , 0.80121821 , 0.89386546 , 0.95477358 , 0.98662733 , 0.99942558 , 0.99988676 , 0.98905127 , 0.95874124 , 0.90094667 , 0.81113021 , 0.6999944 , 0.59367968 , 0.52614263};
+__constant float HFB_BP[16] = { 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f};
+
+#define BIT_REVERSE_7_BITS(index) ((( ( (((index) * 0x0802) & 0x22110) | (((index) * 0x8020)&0x88440) ) * 0x10101 ) >> 17) & 0x7F)
+//input data is float2 with beam-pol-time, try to do 3 N=128 at once so that we can sum 3 time samples
+//LWS = {     64 ,  1  } 
+//GWS = {nsamp/6*, 1024} 
+
+__kernel void upchannelize(__global float2 *data, __global float *results_array, __global float *hfb_output_array){
```

Comment:
I realize the original kernel likely doesn't have comments either, but it would be nice to say at least how this one differs from the original.